server/src/test/java/org/elasticsearch/search/profile/query/ProfileCollectorManagerTests.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.profile.query;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.sandbox.search.ProfilerCollectorResult;
+import org.apache.lucene.search.Collector;
+import org.apache.lucene.search.CollectorManager;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TopScoreDocCollector;
+import org.apache.lucene.search.similarities.BM25Similarity;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.tests.index.RandomIndexWriter;
+import org.apache.lucene.tests.search.DummyTotalHitCountCollector;
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class ProfileCollectorManagerTests extends ESTestCase {
+
+    private static class TestCollector extends DummyTotalHitCountCollector {
+
+        private final int id;
+
+        TestCollector(int id) {
+            this.id = id;
+        }
+    }
+
+    public void testNewCollector() throws IOException {
```

Comment:
I am thinking that testing newCollector and reduce separately is a bit overkill, if we test a search with an index searcher that calls them both like we do below. What do you think?

Reply:
I joined the two first tests into one and removed the getResultTree() test in favour of the one with the index searcher. I think its still good to have one "stupid" test just for newCollector() and reduce(). It communicates its intent more clearly. I also added javadocs to the tests because I often find it hard to remember the intent of these tests later.